### PR TITLE
Add tagging tool

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: tools/tagging.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        self._input_directory = os.path.join(ASAPP_ROOT, 'data', self._client, self._start_date)
+
+    @property
+    def input_file(self):
+        return os.path.join(
+            self._input_directory,
+            f'{CLIENT_FULL_NAMES[self._client]}-tagsource{self._start_date}.csv'
+        )
+
+    def run(self):
```
Comment (lisasapp) [inline tools/tagging.py]: if run is almost always going to run _validate_input() and _run_steps() it should be implemented in the base class

File: tools/base.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from abc import ABCMeta, abstractmethod
+
+
+class BaseTool(metaclass=ABCMeta):
+
+    @property
+    @abstractmethod
+    def input_file(self):
```
Comment (lisasapp) [inline tools/base.py]: I don't think this will be relevant for a lot of the tools

File: tools/__init__.py [hunk 3]
```diff
@@ -4,3 +4,9 @@
 ASAPP_ROOT = config.env_vars['ASAPP_ROOT']
 ASAPP_PRODML_ROOT = config.env_vars['ASAPP_PRODML_ROOT']
 ASAPP_MLENG_ROOT = config.env_vars['ASAPP_MLENG_ROOT']
+
+
+CLIENT_FULL_NAMES = {
+    'condor': 'comcast',
+    'spear': 'sprint'
+}
```
Comment (lisasapp) [inline tools/__init__.py:12]: again with this, I don't think constants should be in the __init__.py file. 

I'm fine with using it either of these ways:
`import constants; some_func(constants.AWFUL)`
`from constants import *; some_func(AWFUL)`

https://stackoverflow.com/questions/6345840/whats-the-best-way-to-initialise-and-use-contants-across-python-classes

**Review by lisasapp — CHANGES_REQUESTED** [review]

File: tools/__init__.py (hunk 3 above)
Comment (ghost) [reply tools/__init__.py:12]: Changed.

**Pull request merged**